plutus-use-cases/src/Plutus/Contracts/PubKey.hs

```diff
@@ -92,11 +93,28 @@ pubKeyContract pk vl = mapError (review _PubKeyError   ) $ do
             -- to update it's database with the new confirmed transaction.
             -- Ultimately, the solution is to move indexed information by the
             -- PAB to the chain-index, so that we get a single source of truth.
-            -- The `waitNSlots 1` only works if you have a chain-index closely
-            -- synced to the local node. If the chain-index is not synced with
-            -- the local node, `unspentTxOutFromRef outRef` will always return
-            -- `Nothing`.
-            void $ waitNSlots 1
+            -- The `awaitChainIndexSlot` blocks the contract until the chain-index
+            -- is synced until the current slot. This is not a good solution,
+            -- as the chain-index is always some time behind the current slot.
```

Comment:
> I don't get this explanation, we have no choice but to wait until the chain-index is up to date right?

That's right, I'll rephrase it as follows:

"The temporary solution is to use the `awaitChainIndexSlot` call which waits until the chain-index is up to date. Meaning, the chain-index's synced slot should be at least as high as the current slot. See https://plutus-apps.readthedocs.io/en/latest/adr/0002-pab-indexing-solution-integration.html" for the full explanation.

Is that okay?

> Also, shouldn't it be awaitChainIndexSlot (slot + 1)?

We could do it that way just in case, but just `awaitChainIndexSlot slot` still guaranties to get the correct result because we request the current slot right after the tx is confirmed. So once the chain-index synced to that slot, the information should be indexed.